### PR TITLE
Milestone 3: Add association

### PR DESCRIPTION
**Pull request by Rachelwebdev**
### In this Milestone, I;

- Created a class Classroom with @label instance variable
- Created the has-many/belongs-to relationship between Classroom and Student.
- Created a class Book with @title and @author instance variables
- Created a class Rental with @date instance variables
- Created the many-to-many relationship between Person and Book
- Fixed linter errors

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title:, author:)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (omarsalem7) [inline book.rb:8]: - [ ] Kinldy considers adding a method for adding rentals to achieve a `many-to-many` relation between `Person` and `Book`  using the intermediate class `Rental`. 

```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+  attr_accessor :name, :age, :parent_permission, :rentals
 
   def initialize(age:, name: 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
```
Comment (omarsalem7) [inline person.rb:14]: - [ ] As stated before kindly add the `add_rental` method also here 
```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Rachelwebdev 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (Rachelwebdev) [reply person.rb:14]: Thank you for the review, I will implement this accordingly

**Review by Whoistolu — APPROVED** [review]
### Hi @Rachelwebdev 

Great job implementing the recommended fixes by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Rachelwebdev** [comment]
Thanks for the approval 👍

**Pull request merged**